Sources/SwiftlyCore/SwiftlyCore.swift

```diff
@@ -1,6 +1,6 @@
 import Foundation
```

Comment:
These SwiftlyVersion changes seem unrelated to the proposed changes to the design, can we omit them from this PR? We can open another PR for them.

Reply:
It is meant to be foundational work for the upgrade detection scenarios in the design. I can raise a separate PR if you feel it would be better to get it in right away.